File: book/python_algorithm_interview/problems/05.py
import collections
from typing import List


def group(strs: List[str]) -> List[List[str]]:
    anagrams = collections.defaultdict(list)

    for word in strs:
        anagrams[''.join(sorted(word))].append(word)
    return list(anagrams.values())


print(group(["eat", "tea", "tan", "ate", "nat", "bat"]))

File: achieve/book/python_algorithm_interview/js/05.ts
function group(strs: string[]): Array<Array<string>> {
  let anagram = {};

  for (let word of strs) {
    const key = word.split("").sort().join("");
    anagram[key] = [...(anagram[key] || []), word];
  }

  return Object.values(anagram);
}

console.log(group(["eat", "tea", "tan", "ate", "nat", "bat"]));
